### PR TITLE
[USMON-902] TLS: add manual attach detach for tls progs

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Introduces endpoints to manually attach or detach from a TLS program.

### Motivation

Extend our capabilities to investigate a remote incident and issue with USM TLS monitoring.

### Additional Notes

List all traced PIDs - 
```bash
sudo curl -s --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/usm/traced_programs | jq
```

Attach - 
```bash
sudo curl -s --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/usm/attach-pid -d '{"pid": 2859, "type": "native"}'
```

Detach -
```bash
sudo curl -s --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/usm/detach-pid -d '{"pid": 2859, "type": "native"}'
```

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]
For NPM owned files

File: cmd/system-probe/modules/network_tracer.go [hunk 1]
```diff
@@ -248,7 +248,9 @@ func (nt *networkTracer) Register(httpMux *module.Router) error {
 	})
 
 	httpMux.HandleFunc("/debug/usm_telemetry", telemetry.Handler)
-	httpMux.HandleFunc("/debug/usm_traced_programs", usm.TracedProgramsEndpoint)
+	httpMux.HandleFunc("/debug/usm/tls/traced_programs", usm.TracedProgramsEndpoint)
+	httpMux.HandleFunc("/debug/usm/tls/attach-pid", usm.AttachPIDEndpoint)
+	httpMux.HandleFunc("/debug/usm/tls/detach-pid", usm.DetachPIDEndpoint)
```
Comment (p-lambert) [inline cmd/system-probe/modules/network_tracer.go:253]: nit: as mentioned IRL I don't think we need to make this change TLS-specific

File: pkg/network/usm/ebpf_gotls.go [hunk 2]
```diff
@@ -307,9 +308,28 @@ func registerCBCreator(mgr *manager.Manager, offsetsDataMap *ebpf.Map, probeIDs
...
+	_ = p.AttachPID(pid)
+}
+
+var (
+	errSelfExcluded                = errors.New("self-excluded")
+	errInternalDDogProcessRejected = errors.New("internal datadog process rejected")
+)
+
+// AttachPID attaches the provided PID to the eBPF program.
+func (p *goTLSProgram) AttachPID(pid uint32) error {
```
Comment (p-lambert) [inline pkg/network/usm/ebpf_gotls.go:330]: nit: move all public methods to the top and leave the private ones at the bottom

File: pkg/network/usm/ebpf_gotls.go [hunk 3]
```diff
@@ -307,9 +308,28 @@ func registerCBCreator(mgr *manager.Manager, offsetsDataMap *ebpf.Map, probeIDs
...
+}
+
 func (p *goTLSProgram) handleProcessStart(pid pid) {
+	_ = p.AttachPID(pid)
+}
+
+var (
+	errSelfExcluded                = errors.New("self-excluded")
+	errInternalDDogProcessRejected = errors.New("internal datadog process rejected")
+)
```
Comment (p-lambert) [inline pkg/network/usm/ebpf_gotls.go:327]: are those returned by public methods? should be part of the public API as well?

File: pkg/network/usm/utils/debugger.go [hunk 4]
```diff
@@ -92,6 +104,95 @@ func (d *fileRegistryDebugger) GetTracedPrograms() []TracedProgram {
...
+
+// AttachPIDEndpoint attaches a PID to an eBPF program.
+func AttachPIDEndpoint(w http.ResponseWriter, r *http.Request) {
+	debugger.runAttacherCallback(w, r, attach)
+}
+
+// DetachPIDEndpoint detaches a PID from an eBPF program.
+func DetachPIDEndpoint(w http.ResponseWriter, r *http.Request) {
+	debugger.runAttacherCallback(w, r, detach)
+}
```
Comment (p-lambert) [inline pkg/network/usm/utils/debugger.go:192]: nice 👍

File: pkg/network/usm/ebpf_gotls.go [hunk 5]
```diff
@@ -307,9 +308,28 @@ func registerCBCreator(mgr *manager.Manager, offsetsDataMap *ebpf.Map, probeIDs
 	}
 }
 
+// DetachPID detaches the provided PID from the eBPF program.
+func (p *goTLSProgram) DetachPID(pid uint32) error {
+	return p.registry.Unregister(pid)
+}
```
Comment (p-lambert) [inline pkg/network/usm/ebpf_gotls.go:314]: I would add something like
```
var _ util.Attacher = &goTLSProgram{}
``` 

To the top of every protocol that implements the interface.
IMO that helps to convey to reader that these methods are part of an interface and also gives you a compilation-time check if the interface ever changes.

**Review by p-lambert — APPROVED** [review]

**Review by p-lambert — APPROVED** [review]
Left a few nit comments, but otherwise looks great

File: cmd/system-probe/modules/network_tracer.go (hunk 1 above)
Comment (guyarb) [reply cmd/system-probe/modules/network_tracer.go:253]: What do you mean?
Today, only tls is scoped to a pid, I cannot attach or detach from plain http or plain kafka

Comment (p-lambert) [reply cmd/system-probe/modules/network_tracer.go:253]: I would just avoid mentioning TLS. 
```
debug/usm/traced_programs
debug/usm/attach_pid
``` 
etc

Comment (guyarb) [reply cmd/system-probe/modules/network_tracer.go:253]: done

**Pull request merged**